content/guides/javascript/get-program-accounts.md

```diff
@@ -0,0 +1,307 @@
+any field stored on our account. Specifically, we can query only for accounts
+that match a particular set of bytes at a particular position. `memcmp` requires
+two arguments:
+
+- `offset`: The position at which to begin comparing data. This position is
+  measured in bytes and is expressed as an integer.
+- `bytes`: The data that should match the account's data. This is represented as
+  a base-58 encoded string should be limited to less than 129 bytes.
+
+It's important to note that `memcmp` will only return results that are an exact
+match on `bytes`. Currently, it does not support comparisons for values that are
+less than or greater than the `bytes` we provide.
+
+In keeping with our Token Program example, we can amend our query to only return
+token accounts that are owned by our wallet address. When looking at a token
+account, we can see the first two fields stored on a token account are both
+pubkeys, and that each pubkey is 32 bytes in length. Given that `owner` is the
+second field, we should begin our `memcmp` at an `offset` of 32 bytes. From
+here, we’ll be looking for accounts whose owner field matches our wallet
+address.
+
+![Account Size](/public/assets/guides/get-program-accounts/memcmp.png)
+
+We can invoke this query via the following example:
+
+```typescript filename="get-program-accounts.ts"
+import { TOKEN_PROGRAM_ID } from "@solana/spl-token";
+import { clusterApiUrl, Connection } from "@solana/web3.js";
+
+(async () => {
+  const MY_WALLET_ADDRESS = "FriELggez2Dy3phZeHHAdpcoEXkKQVkv6tx3zDtCVP8T";
+  const connection = new Connection(clusterApiUrl("devnet"), "confirmed");
+
+  const accounts = await connection.getParsedProgramAccounts(
+    TOKEN_PROGRAM_ID, // new PublicKey("TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA")
+    {
+      filters: [
+        {
+          dataSize: 165, // number of bytes
+        },
+        {
+          memcmp: {
+            offset: 32, // number of bytes
+            bytes: MY_WALLET_ADDRESS, // base58 encoded string
+          },
+        },
+      ],
+    },
+  );
+
+  console.log(
+    `Found ${accounts.length} token account(s) for wallet ${MY_WALLET_ADDRESS}: `,
+  );
+  accounts.forEach((account, i) => {
+    console.log(
+      `-- Token Account Address ${i + 1}: ${account.pubkey.toString()} --`,
+    );
+    console.log(`Mint: ${account.account.data["parsed"]["info"]["mint"]}`);
+    console.log(
+      `Amount: ${account.account.data["parsed"]["info"]["tokenAmount"]["uiAmount"]}`,
+    );
+  });
+  /*
+    // Output
+
+    Found 2 token account(s) for wallet FriELggez2Dy3phZeHHAdpcoEXkKQVkv6tx3zDtCVP8T: 
+    -- Token Account Address 0:  H12yCcKLHFJFfohkeKiN8v3zgaLnUMwRcnJTyB4igAsy --
+    Mint: CKKDsBT6KiT4GDKs3e39Ue9tDkhuGUKM3cC2a7pmV9YK
+    Amount: 1
+    -- Token Account Address 1:  Et3bNDxe2wP1yE5ao6mMvUByQUHg8nZTndpJNvfKLdCb --
+    Mint: BUGuuhPsHpk8YZrL2GctsCtXGneL1gmT5zYb7eMHZDWf
+    Amount: 3
+  */
+})();
+```
+
```

Comment:
Remove the output from the script, and instead add:

___

The output should look like: 
```
    Found 2 token account(s) for wallet FriELggez2Dy3phZeHHAdpcoEXkKQVkv6tx3zDtCVP8T: 
    -- Token Account Address 0:  H12yCcKLHFJFfohkeKiN8v3zgaLnUMwRcnJTyB4igAsy --
    Mint: CKKDsBT6KiT4GDKs3e39Ue9tDkhuGUKM3cC2a7pmV9YK
    Amount: 1
    -- Token Account Address 1:  Et3bNDxe2wP1yE5ao6mMvUByQUHg8nZTndpJNvfKLdCb --
    Mint: BUGuuhPsHpk8YZrL2GctsCtXGneL1gmT5zYb7eMHZDWf
    Amount: 3
```